pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/tests/Test_TNCuda_Var.cpp

```diff
@@ -269,16 +309,21 @@ TEMPLATE_TEST_CASE("Test var value of HamiltonianObs", "[MPSTNCuda_Var]", float,
         std::size_t bondDim = GENERATE(2, 3, 4, 5);
         std::size_t num_qubits = 3;
         std::size_t maxBondDim = bondDim;
+        DevTag<int> dev_tag{0, 0};
 
-        TensorNetT mps_state{num_qubits, maxBondDim};
+        std::unique_ptr<TNDeviceT> tn_state =
+            createTNState<TNDeviceT>(num_qubits, maxBondDim, dev_tag);
 
-        mps_state.applyOperations(
+        tn_state->applyOperations(
             {{"RX"}, {"RY"}, {"RX"}, {"RY"}}, {{0}, {0}, {1}, {1}},
             {{false}, {false}, {false}, {false}}, {{0.5}, {0.5}, {0.2}, {0.2}});
 
-        mps_state.append_mps_final_state();
+        if constexpr (std::is_same_v<TNDeviceT, MPSTNCuda<double>> ||
```

Comment:
Could you try if `tn_state->getMethod() == "mps"` works here?

Reply:
I think it will be better to keep it at compile time. What do you think?  :thinking:

Reply:
I think we would also like to explicitly test `getMethod()` right?

Reply:
ok I will change it

Reply:
ok, feel free to keep the previous implementation

Reply:
Sorry, I don't understand where to add the `PL_ABORT` :sweat_smile:

Reply:
PL_ABORT(tn_state->getMethod() == "mps", "error messafe");
       tn_state->append_mps_final_state();

Reply:
Done!
.